src/managers/TileManager.ts

```diff
@@ -38,16 +41,78 @@ class TileManager {
                 ),
                 session.seed,
             );
-            this.list.forEach((tile: Tile) => {
-                const state: TileState = TileState.Unused; // TODO
+
+            for (let i: int = 0; i < this.list.length; i++) {
+                const tile: Tile = this.list[i];
                 const dto: Nullable<TileDTO> =
                     dtos.find((dto: TileDTO) => dto.id === tile.id) || null;
+
+                const state: TileState = this.determineState(
+                    dto,
+                    session.turnPlayer,
+                    i,
+                    initialAmount
+                );
                 tile.apply(state, dto);
-            });
+            }
             this.onSync.emit();
         });
     }
 
+    private determineState(
+        dto: Nullable<TileDTO>,
+        turnPlayer: Nullable<int>,
+        index: int,
+        initialAmount: int
+    ): TileState {
+        
+
+        if (dto !== null) {
+            return TileState.Placed;
+        }
+        assert(turnPlayer !== null);
```

Comment:
this leads to en assertion error while in lobby and the game isnt started yet so i would replace the assertion with:
```
if (turnPlayer === null) {
    return TileState.Unused;
}
```

(and it will not work yet because the backend needs to set the turnplayer to 0)

Reply:
Wow, great catch.